client/state/sites/test/actions.js

```diff
@@ -215,7 +215,9 @@ describe( 'actions', () => {
 
 		test( 'should dispatch receive deleted site when request completes', async () => {
 			await deleteSite( 2916284 )( spy, getState );
-			expect( spy ).toHaveBeenCalledWith( receiveDeletedSite( 2916284 ) );
+			const expected = receiveDeletedSite( 2916284 ).toString();
+			const received = spy.mock.calls[ 1 ].toString();
```

Comment:
This is a bit sloppy but will do the job instead of having to move actions to separate modules in order to be able to mock them properly.

I'm fine to follow up in another PR if you think it's too sloppy 😅

Reply:
This worked for me and it tests the same thing as the original test:
```js
const dispatch = ( action ) => {
  if ( typeof action === 'function' ) {
    return action( dispatch, getState );
  }
  return spy( action );
};
await dispatch( deleteSite( 2916284 ) );
expect( spy ).toHaveBeenCalledWith( { type: 'SITE_DELETE_RECEIVE', siteId: 2916284 } );
```
The `dispatch` function is basically a copy of the `redux-thunk` middleware.